player/command.c

```diff
@@ -5684,39 +5684,39 @@ static void cmd_frame_step(void *p)
 {
     struct mp_cmd_ctx *cmd = p;
     struct MPContext *mpctx = cmd->mpctx;
+    int frames = cmd->args[0].v.i;
+    int flags = cmd->args[1].v.i;
 
-    if (!mpctx->playback_initialized) {
+    if (!mpctx->playback_initialized || frames == 0) {
         cmd->success = false;
         return;
     }
 
-    if (cmd->cmd->is_up_down) {
+    if (frames > 0 && !flags && cmd->cmd->is_up_down) {
         if (cmd->cmd->is_up) {
             if (mpctx->step_frames < 1)
                 set_pause_state(mpctx, true);
         } else {
             if (cmd->cmd->repeated) {
                 set_pause_state(mpctx, false);
             } else {
-                add_step_frame(mpctx, 1);
+                add_step_frame(mpctx, frames, flags);
             }
         }
     } else {
-        add_step_frame(mpctx, 1);
+        add_step_frame(mpctx, frames, flags);
     }
 }
 
 static void cmd_frame_back_step(void *p)
 {
     struct mp_cmd_ctx *cmd = p;
-    struct MPContext *mpctx = cmd->mpctx;
-
-    if (!mpctx->playback_initialized) {
-        cmd->success = false;
-        return;
-    }
-
-    add_step_frame(mpctx, -1);
+    struct mp_cmd_arg frames = {.v.i = -1};
+    struct mp_cmd_arg flags = {.v.i = 1};
+    cmd->args = talloc_realloc(cmd, cmd->args, struct mp_cmd_arg, 2);
+    cmd->args[0] = frames;
+    cmd->args[1] = flags;
```

Comment:
`cmd->nargs` needs to be set and type in mp_cmd_arg. Also not sure I like this realloc here.

Reply:
It's dumb. I should just make it call `cmd_frame_step`.

Reply:
Done